File: 1092/1092cp2/hw3/main.c
#include <stdio.h>
#include <limits.h>
#define MIN(a,b) (((a)<(b))?(a):(b))

int main(){
	int n;
	scanf("%d", &n);
	int solutions[1000010];
	solutions[0] = 0;
	solutions[1] = 1;
	for(int i = 2; i <= n; ++i){
		solutions[i] = i;
		for(int j = 2; j*j <= i; ++j){
			solutions[i] = MIN(solutions[i], solutions[i - j*j] + 1);
		}
	}
	printf("%d\n", solutions[n]);
}

File: 1092/1092cp2/hw3/main.py
string = "#include <stdio.h>\nint main(){\n    int array[1000000] = {0,"
f = open("test.c", "w")
f.write(string)
solutions = []
solutions.append(0)
for ea in range(1,1000010):
   solutions.append(1000010)
perfect = 1
num = 1
while(perfect <= 1000000):
    for k in range(perfect, 1000001):
        solutions[k] = min(solutions[k], solutions[k - perfect] + 1)
        #print(solutions[k])
    num += 1
    perfect = num*num
    print(perfect)
for i in range(1,1000001):
    if i != 1:
        f.write(",")
    f.write(str(solutions[i]))
f.write("};\n    int N;\n    scanf(\"%d\",&N);\n    printf(\"%d\\n\", array[N]);\n}")
f.close
